crypto/ocsp/ocsp_http.c

```diff
@@ -176,12 +177,20 @@ int OCSP_REQ_CTX_nbio(OCSP_REQ_CTX *rctx) {
 
       OPENSSL_FALLTHROUGH;
     case OHS_ASN1_WRITE_INIT:
-      rctx->asn1_len = BIO_get_mem_data(rctx->mem, NULL);
+      if(!BIO_mem_contents(rctx->mem, NULL, &tmp_data_len)) {
+        rctx->state = OHS_ERROR;
+        return 0;
+      }
+      rctx->asn1_len = (unsigned long)tmp_data_len;
```

Comment:
size_t's size is platform dependent, but will be the same size or smaller than unsigned long so this is safe.